style/javascript/README.md

```diff
@@ -0,0 +1,24 @@
+JavaScript
+==========
+
+[Sample](sample.js)
+
+* Prefer ES6 classes over library provided classes.
+* Use strict equality checks (`===` and `!==`) except when comparing against
+  (`null` or `undefined`).
+* Prefer [arrow functions] `=>`, over the `function` keyword except when
+  defining classes or methods.
+* Use semicolons at the end of each statement.
+* Prefer double quotes.
+* Use `PascalCase` for classes, `lowerCamelCase` for variables and functions,
+  `SCREAMING_SNAKE_CASE` for constants, `_singleLeadingUnderscore` for private
+  variables and functions.
+* Prefer [template strings] over string concatenation.
+* Prefer promises over callbacks.
+* Prefer array functions like `map` and `forEach` over `for` loops.
+* Use `const` to declare variables.
```

Comment:
Since nothing in JavaScript is truly immutable (except for numbers), I wouldn't go so far as to use `const` (except for actual constants). However, I *would* use `let` over `var` -- the block scope works like Ruby so it's familiar, the block scope also prevents scope issues with things like `for`, and `let`s aren't hoisted to the top of functions so you can't declare them out of order.